### PR TITLE
Greedy explore

### LINKED ISSUE #689: greedy exploration

### PR DESCRIPTION
**Pull request by amburger66**
closes #689

### CODE REVIEW

**Comment by tomsilver** [comment]
I feel like I'm too close to this since we pair programmed it, so when you're ready for a review, let's ask ronuchit

File: scripts/supercloud/run_action_policy_experiments.sh
Comment (ronuchit) [inline scripts/supercloud/run_action_policy_experiments.sh:1]: could you add the results of this experiment to this PR? (can be done after merging too.) it helps keep things organized & lets us easily look back to performance of old things that were tried

File: src/approaches/interactive_learning_approach.py
Comment (ronuchit) [inline src/approaches/interactive_learning_approach.py:282]: we can verify this via `assert not CFG.sampler_learning_use_goals`

File: src/settings.py
Comment (ronuchit) [inline src/settings.py:209]: is there a reason you want to change these defaults? (doesn't matter much, just wondering)

File: src/approaches/interactive_learning_approach.py
Comment (ronuchit) [inline src/approaches/interactive_learning_approach.py:278]: inside this function, you're re-computing all the ground nsrts, but that only has to be done once at the start of this function (using the objects in `init`). so, we should just make _sample_applicable_ground_nsrt take in all the ground  nsrts.

File: src/approaches/interactive_learning_approach.py
Comment (ronuchit) [inline src/approaches/interactive_learning_approach.py:306]: should we `assert not np.isinf(best_score)`?

File: src/approaches/interactive_learning_approach.py
Comment (ronuchit) [inline src/approaches/interactive_learning_approach.py:302]: is there a reason you want to force all the trajectories to be exactly of length `interactive_max_trajectory_length`? what's the problem if they're shorter than that?

**Review by ronuchit — COMMENTED** [review]
looks good!

File: src/settings.py
Comment (amburger66) [reply src/settings.py:209]: the combination of frequency + strict_best_seen is just not very good

File: src/approaches/interactive_learning_approach.py
Comment (ronuchit) [inline src/approaches/interactive_learning_approach.py:310]: oh i actually meant that we should do this assert after the loop, right before (and at the same indentation level as) `act_policy = utils.option_plan_to_policy(best_options)`. the idea is to make sure that we saw at least one trajectory, so best_score isn't just equal to its initialization (`-np.inf`)

File: src/main.py
Comment (ronuchit) [inline src/main.py:222]: are there any others using online stuff that this might affect? @tomsilver

File: src/approaches/interactive_learning_approach.py
Comment (amburger66) [reply src/approaches/interactive_learning_approach.py]: i was thinking about this too... i guess it's not a problem and if it's the best trajectory then we should use it!

File: src/main.py
Comment (tomsilver) [reply src/main.py:222]: yeah Ashay actually added this line too in his open PR, for the same reason as we're adding it here

**Review by ronuchit — APPROVED** [review]
lgtm! remember to add results when they're ready!

**Comment by amburger66** [comment]
coverenv_glib_vs_greedy_lookahead_num_transitions_query_cost
coverenv_glib_vs_greedy_lookahead_num_transitions_perc_solved
coverenv_glib_vs_greedy_lookahead_num_transitions_perc_plan_fail
coverenv_glib_vs_greedy_lookahead_num_transitions_perc_exec_fail

**Comment by amburger66** [comment]
https://user-images.githubusercontent.com/26547725/164309727-e7c39b5a-a8d6-4bbd-a266-38141cd6cf09.mp4

**Pull request merged**
